dipy/align/transforms.pyx

```diff
@@ -0,0 +1,810 @@
+
+def eval_jacobian_function(int ttype, int dim, double[:] theta, double[:] x,
+                           double[:,:] J):
+    r""" Compute the Jacobian of a transformation with given parameters at x
+
+    Parameters
+    ----------
+    ttype : int
+        the type of the transformation (use transform_type dictionary to map
+        transformation name to the associated int )
+    dim : int
+        the domain dimension of the transformation (either 2 or 3)
+    theta : array, shape (n,)
+        the parameters of the transformation at which to evaluate the Jacobian
+    x : array, shape (dim,)
+        the point at which to evaluate the Jacobian
+    J : array, shape (dim, n)
+        the destination matrix for the Jacobian
+    """
+    with nogil:
+        get_jacobian_function(ttype, dim)(theta, x, J)
+
+
+def param_to_matrix(int ttype, int dim, double[:] theta, double[:,:] T):
+    r""" Compute the matrix associated to the given transform and parameters
+
+    Parameters
+    ----------
+    ttype : int
+        the type of the transformation (use transform_type dictionary to map
+        transformation name to the associated int )
+    dim : int
+        the domain dimension of the transformation (either 2 or 3)
+    theta : array, shape (n,)
+        the transformation parameters
+    T : array, shape (dim + 1, dim + 1)
+        the buffer to write the transform matrix
+    """
+    with nogil:
+        get_param_to_matrix_function(ttype, dim)(theta, T)
+
+
+def get_identity_parameters(int ttype, int dim, double[:] theta):
+    r""" Gets the parameters corresponding to the identity transform
+
+    Parameters
+    ----------
+    ttype : int
+        the type of the transformation (use transform_type dictionary to map
+        transformation name to the associated int )
+    dim : int
+        the domain dimension of the transformation (either 2 or 3)
+    theta : array, shape (n,)
+        the buffer to write the identity parameters into
+    """
+    _get_identity_parameters(ttype, dim, theta)
+
+
+cdef void _get_identity_parameters(int ttype, int dim, double[:] theta) nogil:
+    if dim == 2:
+        if ttype == TRANSLATION:
+            theta[:2] = 0
+        elif ttype == ROTATION:
+            theta[0] = 0
+        elif ttype == RIGID:
+            theta[:3] = 0
+        elif ttype == SCALING:
+            theta[0] = 1
+        elif ttype == AFFINE:
+            theta[0], theta[1], theta[2] = 1, 0, 0
+            theta[3], theta[4], theta[5] = 0, 1, 0
+    elif dim == 3:
+        if ttype == TRANSLATION:
+            theta[:3] = 0
+        elif ttype == ROTATION:
+            theta[:3] = 0
+        elif ttype == RIGID:
+            theta[:6] = 0
+        elif ttype == SCALING:
+            theta[0] = 1
+        elif ttype == AFFINE:
+            theta[0], theta[1], theta[2], theta[3] = 1, 0, 0, 0
+            theta[4], theta[5], theta[6], theta[7] = 0, 1, 0, 0
+            theta[8], theta[9], theta[10], theta[11] = 0, 0, 1, 0
+
```

Comment:
The def's is now in charge of passing correct arrays to the cdef's or raise a ValueError if the request was invalid